### DISCUSSION TITLE
Basic question about reulermultinom.

### DISCUSSION THREAD

**Discussion by kingaa** [kingaa/pomp#187] [Q&A] [answered]
@mbennun writes:

I am new to POMP and have a very basic question about the example code attached below. From the S state there are two rates that should be sampled using 'reulermultinom', rate[1] and rate[2]. Similarly, for the I states the rates are rate[3] and rate[4]. In each case the output should be two integers sampled from the proper size (S or I) and the rates (1 and 2 or 3 and 4)
In each case I was expecting the call to look e.g, reulermultinom(1, S, &rate[1], dt, &dN[1]). Doesn't this call:
reulermultinom(2, I, &rate[3], dt, &dN[3]) return two randome variables for each rate?
It is also not really clear to me how dN[2]and dn[4] are calculated.
Thanks in advance.
```
sir.step <- "
double rate[6];
double dN[6];
double P;
P = S + I + R;
rate[0] = mu * P; // birth
rate[1] = Beta * I / P; // transmission
rate[2] = mu; // death from S
rate[3] = gamma; // recovery
rate[4] = mu; // death from I
rate[5] = mu; // death from R
dN[0] = rpois(rate[0] * dt);
reulermultinom(2, S, &rate[1], dt, &dN[1]);
reulermultinom(2, I, &rate[3], dt, &dN[3]);
reulermultinom(1, R, &rate[5], dt, &dN[5]);
S += dN[0] - dN[1] - dN[2];
I += dN[1] - dN[3] - dN[4];
R += dN[3] - dN[5];
H += dN[1];
```

**Comment by kingaa** [comment, marked as answer]
@mbennun:

Have a look at [FAQ 3.6](https://kingaa.github.io/pomp/FAQ.html#eulermultinomial-approximation) and see if it doesn't answer your question.  There is also [this documentation](https://kingaa.github.io/pomp/C_API.html#euler-multinomial-distribution) on the C API to the `reulermultinom` function.  

If you are not familiar with C pointers, you may need to read up on the concept, for example [here](https://www.tutorialspoint.com/cprogramming/c_pointers.htm), [here](https://www.w3schools.com/c/c_pointers.php), or [here](https://en.wikipedia.org/wiki/Pointer_(computer_programming)).

**Comment by mbennun** [comment]
Thank you! FAQ 3.6 answers my question very clearly.
